File: SW/D2_1979.py
def check(i):
    global res
    row = "".join(board[i]).split('0')
    for v in row:
        if len(v) == K:
            res += 1
            
for t in range(1, int(input())+1):
    N, K = map(int, input().split())
    board = [list(input().split()) for _ in range(N)]

    res = 0
    for i in range(N):
        check(i)
    
    board = list(zip(*board))
    for i in range(N):
        check(i)
    
    print("#{} {}".format(t, res))

File: SW/D2_1979.java
import java.util.*;

public class D2_1979 {
    static int N, K;
    static int[][] board;
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int T = sc.nextInt();
        for (int t=1; t<=T; t++) {
            N = sc.nextInt();
            K = sc.nextInt();
            
            board = new int[N][N];
            int[] check = new int[N];
            for (int i=0; i<N; i++) {
                for (int j=0; j<N; j++) {
                    board[i][j] = sc.nextInt();
                }
            }

            int res = 0;
            for (int i=0; i<N; i++) {
                int cnt = 0;
                for (int j=0; j<N; j++) {
                    if (board[i][j] == 1) {
                        cnt += 1;
                        check[j] = cnt;
                    } else {
                        cnt = 0;
                        check[j] = 0;
                    }
                }

                for (int j : check) {
                    System.out.println(j);
                }
            }

            
        }
    }    
}
